assets/js/components/user-input/UserInputPreviewGroup.js

```diff
@@ -266,81 +257,28 @@ export default function UserInputPreviewGroup( {
 			</LoadingWrapper>
 
 			{ ! isEditing && (
-				<div className="googlesitekit-user-input__preview-answers">
-					<LoadingWrapper
-						loading={ loading }
-						width="340px"
-						height="36px"
-					>
-						{ errorMessage && (
-							<p className="googlesitekit-error-text">
-								{ errorMessage }
-							</p>
-						) }
-
-						{ ! errorMessage &&
-							values.map( ( value ) => (
-								<div
-									key={ value }
-									className="googlesitekit-user-input__preview-answer"
-								>
-									{ options[ value ] }
-								</div>
-							) ) }
-					</LoadingWrapper>
-				</div>
+				<UserInputPreviewAnswers
+					values={ values }
+					options={ options }
+					loading={ loading }
+					errorMessage={ errorMessage }
+				/>
 			) }
-
 			{ isEditing && (
-				<Fragment>
-					<UserInputSelectOptions
-						slug={ slug }
-						max={ USER_INPUT_MAX_ANSWERS[ slug ] }
-						options={ options }
-						alignLeftOptions
-						descriptions={ USER_INPUT_ANSWERS_PURPOSE_DESCRIPTIONS }
-					/>
-					{ errorMessage && (
-						<p className="googlesitekit-error-text">
-							{ errorMessage }
-						</p>
-					) }
-					{ settingsView && (
-						<Fragment>
-							<UserInputQuestionAuthor slug={ slug } />
-
-							{ saveSettingsError && (
-								<ErrorNotice error={ saveSettingsError } />
-							) }
-
-							<div className="googlesitekit-user-input__preview-actions">
-								<SpinnerButton
-									disabled={ answerHasError }
-									onClick={
-										hasSettingChanged
-											? submitChanges
-											: toggleEditMode
-									}
-									isSaving={ isScreenLoading }
-								>
-									{ hasSettingChanged || isSavingSettings
-										? __(
-												'Apply changes',
-												'google-site-kit'
-										  )
-										: __( 'Save', 'google-site-kit' ) }
-								</SpinnerButton>
-								<Button
-									tertiary
-									disabled={ isScreenLoading }
-									onClick={ handleOnCancelClick }
-								>
-									{ __( 'Cancel', 'google-site-kit' ) }
-								</Button>
-							</div>
-						</Fragment>
-					) }
-				</Fragment>
+				<UserInputEditModeContent
+					slug={ slug }
+					options={ options }
+					errorMessage={ errorMessage }
+					isSavingSettings={ isSavingSettings }
+					isScreenLoading={ isScreenLoading }
+					saveSettingsError={ saveSettingsError }
+					answerHasError={ answerHasError }
+					settingsView={ settingsView }
+					hasSettingChanged={ hasSettingChanged }
+					submitChanges={ submitChanges }
+					toggleEditMode={ toggleEditMode }
+					handleOnCancelClick={ handleOnCancelClick }
```

Comment:
All of the selectors/actions/functions passed through props here (excluding `settingsView` which is passed as a prop to this component) are only used within `UserInputEditModeContent` so they can be moved within this component to reduce the number of props passed here and isolate the logic where it's needed.